v2/handler_test.go

```diff
@@ -0,0 +1,205 @@
+package btclog
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"github.com/btcsuite/btclog"
+	"io"
+	"testing"
+	"time"
+)
+
+// TestDefaultHandler tests that the DefaultHandler's output looks as expected.
+func TestDefaultHandler(t *testing.T) {
+	t.Parallel()
+
+	timeSource := func() time.Time {
+		return time.Unix(100, 100)
+	}
+
+	tests := []struct {
+		name               string
+		handlerConstructor func(w io.Writer) Handler
+		level              btclog.Level
+		logFunc            func(log Logger)
+		expectedLog        string
+	}{
+		{
+			name: "Basic calls and levels",
+			handlerConstructor: func(w io.Writer) Handler {
+				return NewDefaultHandler(
+					w, WithTimeSource(timeSource),
+				)
+			},
+			level: LevelDebug,
+			logFunc: func(log Logger) {
+				log.Info("Test Basic Log")
+				log.Debugf("Test basic log with %s", "format")
+				log.Trace("Log should not appear due to level")
+			},
+			expectedLog: `1970-01-01 02:01:40.000 [INF]: Test Basic Log
+1970-01-01 02:01:40.000 [DBG]: Test basic log with format
+`,
+		},
+		{
+			name: "Call site",
+			handlerConstructor: func(w io.Writer) Handler {
+				return NewDefaultHandler(
+					w, WithNoTimestamp(),
+					WithCallSiteSkipDepth(7),
+					WithCallerFlags(Lshortfile),
+				)
+			},
+			level: LevelInfo,
+			logFunc: func(log Logger) {
+				log.Info("Test Basic Log")
+			},
+			expectedLog: `[INF] handler_test.go:196: Test Basic Log
+`,
+		},
+		{
+			name: "Sub-system tag",
+			handlerConstructor: func(w io.Writer) Handler {
+				h := NewDefaultHandler(w, WithNoTimestamp())
+				return h.SubSystem("SUBS")
+			},
+			level: LevelInfo,
+			logFunc: func(log Logger) {
+				log.Info("Test Basic Log")
+			},
+			expectedLog: `[INF] SUBS: Test Basic Log
+`,
+		},
+		{
+			name: "Test all levels",
+			handlerConstructor: func(w io.Writer) Handler {
+				return NewDefaultHandler(w, WithNoTimestamp())
+			},
+			level: LevelTrace,
+			logFunc: func(log Logger) {
+				log.Trace("Trace")
+				log.Debug("Debug")
+				log.Info("Info")
+				log.Warn("Warn")
+				log.Error("Error")
+				log.Critical("Critical")
+			},
+			expectedLog: `[TRC]: Trace
+[DBG]: Debug
```

Comment:
hmm formatting seems weird here.

Reply:
no real other way to define this such that things match on a char by char level. It's just a test too :)